app/models/concerns/hostext/search.rb

```diff
@@ -216,6 +216,8 @@ def param_conditions(p)
               conditions << "hosts.hostgroup_id IN (#{param.hostgroup.subtree_ids.join(', ')})"
             when 'HostParameter'
               conditions << "hosts.id = #{param.reference_id}"
+            when 'SubnetParameter'
+              conditions << "hosts.subnet_id = #{param.reference_id}"
```

Comment:
@sean797 the bug was fixed and merged: https://github.com/theforeman/foreman/commit/f2506d3482098dcd4faffe64c036f139d77034d7
if you change this to `"nics.subnet_id = #{param.reference_id}"` it will work